app/commands/decidim/verify_wo_registration/do_verify_wo_registration.rb

```diff
@@ -20,10 +20,20 @@ def initialize(form)
       def call
         if authorizations_exists?
           if existing_registered_user?
+            # ignore form data, make the participant use the already existing user
             broadcast(:use_registered_user)
           elsif existing_impersonated_user?
-            authorize_user
-            broadcast(:ok, user)
+            # we can not reuse existing authorization because it will raise "A participant is already authorized with the same data." on @form.valid?
+            # as it is an impersonated user, we can safely destroy it and perform the verification process again
+            transaction do
+              destroy_authorization
```

Comment:
If you destroy the authorization, then I can vote for the same proposal multiple times:

1. I use this module to verify myself without registering and get a 30min session. This creates a new user, say user ID 1
2. I vote for the proposal A. Vote is related to user ID 1.
3. I wait until the 30min session has expired.
4. I use this module to verify myself again. Since the verification is destroyed, I can verify myself again and a new user is created, say user ID 2.
5. Since the user is brand new, I can vote for the proposal A again. Vote is now related to user ID 2.

I'd remove this

Reply:
Good catch @mrcasals you're right!
Let me see if I can find a way to bind the new authorization to the previously impersonated user, or otherwise do not allow new sessions for previously impersonated users

Reply:
After reviewing the code, I'm not sure your proposed workflow is correct.
When we destroy the authorization on `destroy_authorization` the previously impersonated user (ID 1) is kept in the command as an attribute. Then, in `authorize_user` we do `create_or_update_authorizations`, we do not create a new impersonated user, we reassign the previously impersonated user (with ID 1) and no new user is created.

So the votes keeps being associated to new impersonated sessions.

Reply:
Although your workflow is may happen if the second workflow is as follows:
1. I use this module to verify myself without registering and get a 30min session. This creates a new user, say user ID 1
2. I vote for the proposal A. Vote is related to user ID 1.
3. An admin resets the authorizations in the middle of the voting period
4. All impersonated users (both via admin or using this module) will be able to verify and vote again via all means: registering, being impersonated in the admin or using this module.

Being said that, the behavior that this module implements is according the requirements of the module and with the principal philosophy of the module which is simplify the steps to the minimal for participants to be able to participate in processes that require being verified to increase participation.

So, as this is a corner case that can happen either the organization has this module enabled or not (so this may happen in any Decidim installation) I think the best will be to add a notice in the reset button to avoid admins reset authorizations in the middle of a process.

Reply:
Ah, you're right, I missed the code in the `existing_impersonated_user?` method because since it ends in `?` I expected to only perform a boolean check, not to add extra logic to the process... 😕 

> An admin resets the authorizations in the middle of the voting period

You're right this can cause issues and this might be out of the scope of this module. Maybe update the README file with a "Warning" section explaining the possible consequences of resetting verifications in the middle of a voting process?